### PR TITLE
Updating the GOES-XRS baseurl following change on noaa side

### LINKED ISSUE #6734: Fido change of URL for GOES15/XRS and earlier data
sunpy/fido returns `urlopen error [Errno 60] Operation timed out` when searching for GOES15 XRS data as NOAA has moved it from 
https://satdat.ngdc.noaa.gov/sem/goes/data/science/xrs/goes15/  to https://www.ncei.noaa.gov/data/goes-space-environment-monitor/access/science/xrs/goes15/ 

GOES08 to 15 now at this new url, and so all these need to be changed. GOES16-18 at different url which hasn't changed.

Fido XRS search returns the shorter cadence "non-science" data versions, i.e. https://www.ncei.noaa.gov/data/goes-space-environment-monitor/access/science/xrs/goes15/gxrs-l2-irrad_science/ then subdirectory YYYY/MM

### PR DESCRIPTION
**Pull request by hayesla**
This fixes #6734.

The GOES-XRS data has now been re-processed for satellites 8-13 too, I'll also add that in this PR

### CODE REVIEW

**Review by dstansby — DISMISSED** [review]
I think this is still a work in progress? I'll mark as needs changes anyway, since this should get a changelog entry.

**Comment by hayesla** [comment]
ah yes, sorry i've two things to update, ill push soon 👍

File: sunpy/net/dataretriever/sources/goes.py [hunk 1]
```diff
@@ -70,8 +70,8 @@ class XRSClient(GenericClient):
     baseurl_old = r'https://umbra.nascom.nasa.gov/goes/fits/%Y/go(\d){2}(\d){6,8}\.fits'
     pattern_old = '{}/fits/{year:4d}/go{SatelliteNumber:02d}{}{month:2d}{day:2d}.fits'
     # GOES XRS 13, 14, 15 from NOAA (re-processed data)
```
Comment (nabobalis) [inline sunpy/net/dataretriever/sources/goes.py:72]: I assume this comment needs updating?

**Comment by nabobalis** [comment]
Since I think this is causing some online fails, I decided to rebase it and update the tests.

Hopefully everything is passing but I will need to double check.

File: sunpy/net/dataretriever/sources/goes.py [hunk 2]
```diff
@@ -21,25 +21,27 @@
...
 
-    * GOES-R 16, 17 :
-        https://data.ngdc.noaa.gov/platforms/solar-space-observing-satellites/goes/goes16/l1b/docs/GOES-R_EXIS_XRS_L1b_Science-Quality_Data_ReadMe.pdf
+    See the following readmes about these data:
+
+    * GOES 1 - 7: https://umbra.nascom.nasa.gov/goes/fits/goes_fits_files_notes.txt
+
+    * Reprocessed 8 - 15: https://www.ncei.noaa.gov/data/goes-space-environment-monitor/access/science/xrs/GOES_1-15_XRS_Science-Quality_Data_Readme.pdf
+
+    * GOES-R 16 - 17: https://data.ngdc.noaa.gov/platforms/solar-space-observing-satellites/goes/goes16/l1b/docs/GOES-R_EXIS_XRS_L1b_Science-Quality_Data_ReadMe.pdf
```
Comment (nabobalis) [inline sunpy/net/dataretriever/sources/goes.py:44]: Do we know when this gets updated for 18? The archive has no data yet. But I saw that SUVI was there.

**Comment by nabobalis** [comment]
Looks like two real online failures outside of some timeouts. One of them doesn't look to be because of this PR but because of SRSClient.

File: sunpy/net/dataretriever/sources/goes.py (hunk 2 above)
Comment (hayesla) [reply sunpy/net/dataretriever/sources/goes.py:44]: no idea on the timelines for this for 18. I'll ask

**Comment by hayesla** [comment]
I could only see one that was related to this PR (unless I missed it), but fixed that one

**Comment by nabobalis** [comment]
> I could only see one that was related to this PR (unless I missed it), but fixed that one

Yeah I think we should fix the others in a separate PR, they arent due to GOES.

**Review by dstansby — COMMENTED** [review]
This looks good now - we should wait before merging to make sure that the online tests are covering the lines changed though. I'll take a look again in half an hour 😄

**Comment by nabobalis** [comment]
Only failed online test is: `FAILED ../../.tox/py39-online/lib/python3.9/site-packages/sunpy/net/hek/tests/test_hek.py::test_mixed_results_get `

**Review by nabobalis — APPROVED** [review]

**Comment by nabobalis** [comment]
Manually backported to 4.0

**Comment by nabobalis** [comment]
Thank you @hayesla for getting this done.

**Pull request merged**